### PR TITLE
[python] Change the `sudo_if` function to avoid issues with Bash

### PR DESCRIPTION
**Pull request by alexander-smolyakov**
**Feature name**:

- ghcr.io/devcontainers/features/python

**Description**:

This PR aims to resolve the `No such file or directory` error when commands are executed via the `sudo_if` function.

_Changelog_:

- Removed double quotes to avoid issues with tokenization;
- Added ability to install the `jupyterlab` package under root user;
- Fixed `jupyterlab` tests;
- Added new tests to cover the "install under root user" scenario;

**Attached related issue**:

- #588

**Checklist**:

- [x] Checked that applied changes work as expected

### CODE REVIEW

File: src/python/install.sh [hunk 1]
```diff
@@ -304,7 +304,7 @@ sudo_if() {
     if [ "$(id -u)" -eq 0 ] && [ "$USERNAME" != "root" ]; then
         su - "$USERNAME" -c "$COMMAND"
     else
-        "$COMMAND"
+        $COMMAND
     fi
```
Comment (samruddhikhandale) [inline src/python/install.sh:308]: Can we fix the `jupyterlab-git` check failures? We'd need to update 👇  to `check "jupyterlab_git" grep jupyterlab_git <<< "$packages" `.

https://github.com/devcontainers/features/blob/1f83ebc28c0b7966726151d8712c785c5c5f4f99/test/python/install_jupyterlab.sh#L20

Same change to https://github.com/devcontainers/features/blob/1f83ebc28c0b7966726151d8712c785c5c5f4f99/test/python/install_additional_jupyterlab.sh#L20

File: src/python/install.sh [hunk 2]
```diff
@@ -304,7 +304,7 @@ sudo_if() {
     if [ "$(id -u)" -eq 0 ] && [ "$USERNAME" != "root" ]; then
         su - "$USERNAME" -c "$COMMAND"
     else
-        "$COMMAND"
+        $COMMAND
```
Comment (samruddhikhandale) [inline src/python/install.sh:307]: Can we bump the minor version? Thanks! https://github.com/devcontainers/features/blob/1f83ebc28c0b7966726151d8712c785c5c5f4f99/src/python/devcontainer-feature.json#L3

File: src/python/install.sh (hunk 1 above)
Comment (alexander-smolyakov) [reply src/python/install.sh:308]: Fixed in https://github.com/devcontainers/features/pull/694/commits/51bec72fd3e2c89d825e831c0c12fb3325d47e12

File: src/python/install.sh (hunk 2 above)
Comment (alexander-smolyakov) [reply src/python/install.sh:307]: Fixed in f4ff7f9920ea62151fc90d9342e7a128042da4fd

[the remark this answers is not in the document]
Comment (alexander-smolyakov) [reply src/python/install.sh:307]: I added the ability to install the jupyterlab package under the root user and implemented tests to cover related changes.

File: src/python/install.sh [hunk 3]
```diff
@@ -304,25 +304,31 @@ sudo_if() {
...
+        sudo_if "${PYTHON_SRC}" -m pip install --upgrade --no-cache-dir "$PACKAGE"
+    else
+        sudo_if "${PYTHON_SRC}" -m pip install --user --upgrade --no-cache-dir "$PACKAGE"
+    fi
 }
 
 add_user_jupyter_config() {
-    CONFIG_DIR="/home/$USERNAME/.jupyter"
-    CONFIG_FILE="$CONFIG_DIR/jupyter_server_config.py"
+    CONFIG_DIR=$1
```
Comment (samruddhikhandale) [inline src/python/install.sh:323]:
```suggestion
    CONFIG_DIR="$1"
```

File: src/python/install.sh [hunk 4]
```diff
@@ -461,13 +467,26 @@ if [ "${INSTALL_JUPYTERLAB}" = "true" ]; then
         exit 1
     fi
 
-    install_user_package jupyterlab
-    install_user_package jupyterlab-git
+    INSTALL_UNDER_ROOT=true;
```
Comment (samruddhikhandale) [inline src/python/install.sh:470]:
```suggestion
    INSTALL_UNDER_ROOT=true
```

**Review by samruddhikhandale — APPROVED** [review]
Left some minor comments, besides that LGTM

**Review by samruddhikhandale — APPROVED** [review]
Left some minor comments, besides that LGTM

File: src/python/install.sh (hunk 3 above)
Comment (alexander-smolyakov) [reply src/python/install.sh]: Fixed in b39c96402c296ff9d158f61741af84d0b07c0003

File: src/python/install.sh (hunk 4 above)
Comment (alexander-smolyakov) [reply src/python/install.sh]: Fixed in b39c96402c296ff9d158f61741af84d0b07c0003

**Review by samruddhikhandale — APPROVED** [review]
Thank you! 🚢

**Pull request merged**